src/@next/SideSheet/SideSheet.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useEffect } from 'react';
+import { Button, PrimaryButton } from '../Button';
+import { ButtonGroup } from '../ButtonGroup';
+import { ComponentAction } from '../../types/componentAction';
+import { Icon } from '../Icon';
+import { Neutral } from '../utilities/colors';
+import { Portal } from '../Portal/Portal';
+import {
+  StyledSideSheetContainer,
+  StyledSideSheetWrapper,
+  StyledSideSheetHeader,
+  StyledButtonContainer,
+  StyledRightContainer,
+  StyledHorizontalLine,
+  StyledSideSheetFooter,
+  StyledButtonGroupContainer,
+} from './SideSheetStyle';
+import { Typography } from '../Typography';
+
+export interface SideSheetProps {
+  isOpen: boolean;
+  title: string;
+  description?: string;
+  showHorizontalLine?: boolean;
+  children?: React.ReactNode;
+  basicButtonAction?: ComponentAction;
+  primaryButtonAction?: ComponentAction;
+  showBackButton?: boolean;
+  onBack?: () => void;
+  showCloseButton?: boolean;
+  onClose?: () => void;
+}
+
+const SideSheet = React.forwardRef<HTMLDivElement, SideSheetProps>(
+  function SideSheet(
+    {
+      isOpen,
+      title,
+      description,
+      showHorizontalLine = true,
+      children,
+      basicButtonAction,
+      primaryButtonAction,
+      showBackButton = true,
+      onBack,
+      showCloseButton = true,
+      onClose,
+      ...props
+    }: SideSheetProps,
+    ref
+  ) {
+    useEffect(() => {
+      if (typeof window === 'undefined' || !window.document) return;
+      const enableScroll = () => (document.body.style.overflow = 'unset');
+      isOpen ? (document.body.style.overflow = 'hidden') : enableScroll();
+      return () => {
+        enableScroll();
+      };
+    }, [isOpen]);
+
+    // fade out effect for 0.2seconds when closed
+    const handleClose = () => {
+      const wrapperDiv = document.getElementById('side-sheet-wrapper-unique');
```

Comment:
Is it not possible doing this via css + react state?

Reply:
let me try, I think I can do something like:
`sidesheet.closed {...` for the css